File: src/node/magma/app/compile/Placeholder.ts
import { Main } from "../../../magma/app/Main";
import { Value } from "../../../magma/app/Value";
import { Option } from "../../../magma/api/option/Option";
import { None } from "../../../magma/api/option/None";
import { Definition } from "../../../magma/app/Definition";
import { Type } from "../../../magma/api/Type";
import { CompileState } from "../../../magma/app/compile/CompileState";
import { Primitive } from "../../../magma/app/Primitive";
export class Placeholder {
	input: string;
	constructor (input: string) {
		this.input = input;
	}
	generate(): string {
		return Main.generatePlaceholder(this.input);
	}
	isFunctional(): boolean {
		return false;
	}
	findChild(): Option<Value> {
		return new None<Value>();
	}
	asDefinition(): Option<Definition> {
		return new None<Definition>();
	}
	toValue(): Option<Value> {
		return new None<Value>();
	}
	resolve(state: CompileState): Type {
		return Primitive.Unknown;
	}
	isVar(): boolean {
		return false;
	}
	generateBeforeName(): string {
		return "";
	}
	generateAsEnumValue(structureName: string): Option<string> {
		return new None<string>();
	}
}

File: src/java/magma/app/compile/Placeholder.java
package magma.app.compile;

import magma.api.Type;
import magma.api.option.None;
import magma.api.option.Option;
import magma.app.Definition;
import magma.app.Main;
import magma.app.Parameter;
import magma.app.Primitive;
import magma.app.Value;

public record Placeholder(String input) implements Parameter, Value, Type {
    @Override
    public String generate() {
        return Main.generatePlaceholder(this.input);
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public Option<Value> findChild() {
        return new None<Value>();
    }

    @Override
    public Option<Definition> asDefinition() {
        return new None<Definition>();
    }

    @Override
    public Option<Value> toValue() {
        return new None<Value>();
    }

    @Override
    public Type resolve(final CompileState state) {
        return Primitive.Unknown;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }

    @Override
    public Option<String> generateAsEnumValue(final String structureName) {
        return new None<String>();
    }
}
